docs/dependency-injection.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Dependency injection in Blazor
+author: rstropek
+description: Learn how Blazor apps can use built-in services by having them injected into components.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/09/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/dependency-injection
+---
+# Dependency injection in Blazor
+
+By [Rainer Stropek](https://www.timecockpit.com)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+Blazor has [dependency injection (DI)](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection) built-in. Apps can use built-in services by having them injected into components. Apps can also define custom services and make them available via DI.
+
+## What is dependency injection?
+
+DI is a technique for accessing services configured in a central location. This can be useful to:
+
+* Share a single instance of a service class across many components (known as a *singleton* service).
+* Decouple components from particular concrete service classes and only reference abstractions. For example, an interface `IDataAccess` is implemented by a concrete class `DataAccess`. When a component uses DI to receive an `IDataAccess` implementation, the component isn't coupled to the concrete type. The implementation can be swapped, perhaps to a mock implementation in unit tests.
+
+Blazor's DI system is responsible for supplying instances of services to components. DI also resolves dependencies recursively so that services themselves can depend on further services. DI is configured during startup of the app. An example is shown later in this topic.
+
+## Add services to DI
+
+After creating a new app, examine the `Main` method in *Program.cs*:
+
+```csharp
+static void Main(string[] args)
+{
+    var serviceProvider = new BrowserServiceProvider(configure =>
```

Comment:
One more thing: I'm proposing that rename `configure` to `services` so that it's consistent with ASP.NET Core.

Reply:
https://github.com/aspnet/Blazor/pull/494